set_bucket_tags/app.py

```diff
@@ -55,31 +59,94 @@ def get_principal_id(tags):
-  syn = synapseclient.Synapse()
+  syn = get_synapse_client()
   user_profile = syn.getUserProfile(synapse_id)
   log.debug(f'Synapse user profile: {user_profile}')
-
   return user_profile
 
 
-def get_synapse_tags(user_profile):
-  '''Derive synapse tags from synapse user profile data'''
-
+def get_ssm_parameter(name):
+  '''Get an parameter from the SSM parameter store'''
+  client = boto3.client('ssm')
+  parameter = client.get_parameter(Name=name)
+  log.debug(f'Synapse ssm parameter: {parameter}')
+  return parameter
+
+
+def get_synapse_team_ids():
+  '''Get synapse team IDs'''
+  TeamToRoleArnMap = os.getenv('TEAM_TO_ROLE_ARN_MAP_PARAM_NAME')
+  ssm_param = get_ssm_parameter(TeamToRoleArnMap)
+  team_to_role_arn_map = json.loads(ssm_param["Parameter"]["Value"])
+  log.debug(f'/service-catalog/TeamToRoleArnMap value: {team_to_role_arn_map}')
+  team_ids = []
+  for item in team_to_role_arn_map:
+    team_ids.append(item["teamId"])
+
+  log.debug(f'Synapse team IDs: {team_ids}')
+  return team_ids
+
+
+def get_synapse_user_team_id(synapse_id, team_ids):
+  '''Get the Synapse team that the user is in
+  :param synapse_id: synapse user id
+  :param team_ids: the synapse team ids
+  :returns: the synapse team id that the user is in, None if user is not
+            in any teams
+  '''
+  syn = get_synapse_client()
+  for team_id in team_ids:
+    membership_status = syn.get_membership_status(synapse_id, team_id)
+    if membership_status["isMember"]:
+      log.debug(f'Synapse user team ID: {team_id}')
+      return team_id
+
+  return None
+
+def get_synapse_user_profile_tags(user_profile, ignore_keys=["createdOn"]):
+  '''Derive synapse tags from synapse user profile data
+  :param user_profile: the synapse user profile info
+  :param ignore_keys: the keys from the user profile to ignore
+         Note - no email tags are returned if userName is ignored
+  :return a list containing a dictionary of tags
+  '''
   tags = []
-  IGNORE_KEYS = ["createdOn"]
   for key, value in user_profile.items():
-    if key in IGNORE_KEYS:
+    if key in ignore_keys:
       continue
 
     # derive synapse email tag based on userName
     if key == "userName":
       synapse_email = f'{value}@synapse.org'
-      tags.append({'Key': 'synapse:email', 'Value': synapse_email})
+      tags.append({'Key': f'{SYNAPSE_TAG_PREFIX}:email', 'Value': synapse_email})
       tags.append({'Key': 'OwnerEmail', 'Value': synapse_email})  # legacy
 
-    tag = {'Key': f'synapse:{key}', 'Value': value}
+    tag = {'Key': f'{SYNAPSE_TAG_PREFIX}:{key}', 'Value': value}
     tags.append(tag)
 
+  log.debug(f'Synapse user profile tags: {tags}')
+  return tags
+
+
+def get_synapse_user_team_tags(synapse_id, synapse_team_ids):
+  '''Derive team tags from synapse team data'''
+  tags = []
+  user_team_id = get_synapse_user_team_id(synapse_id, synapse_team_ids)
+  if user_team_id:
+    tags.append({'Key': f'{SYNAPSE_TAG_PREFIX}:teamId', 'Value': user_team_id})
```

Comment:
This is very minor but are tag name usually camelCase or snake_case?  If the latter, then perhaps this should be "team_id" instead of teamId.

Reply:
i'm using the convention that is established by synapse.  i just take all of the keys from the synapse response and add it as-is.  i'm using `teamId` to match that convention. 

```
> https://repo-prod.prod.sagebase.org/repo/v1/userProfile/3411506
{
  "createdOn": "2020-06-18T16:34:18.000Z",
  "firstName": "khai",
  "lastName": "do",
  "ownerId": "3411506",
  "userName": "khaido1"
}```